module status update without force package package profile

swadeley opened the pull request:
Hello

Test title is rather cumbersome. Test is adapted from Omkar's tests above. Mainly `test_module_status_update_from_content_host_to_satellite`.

I need to add a wait for event line.

omkarkhatavkar commented on tests/foreman/ui/test_contenthost.py:
```diff
@@ -914,6 +914,75 @@ def test_module_status_update_from_content_host_to_satellite(session, vm_module_
         )
 
+@tier3
+def test_module_status_update_without_force_upload_package_profile(session, vm_module_streams):
```
@swadeley thank you for adding this!

swadeley commented:
Hello

sorry for delay, test is failing:

    >               query={'search': 'name={}'.format(vm.hostname)})
    E           AttributeError: 'function' object has no attribute 'hostname'

    tests/foreman/ui/test_contenthost.py:951: AttributeError

So I ran pytest against the complete `tests/foreman/ui/test_contenthost.py` file and got:
=== 4 failed, 19 passed, 39 warnings, 1 error in 7252.60 seconds ===

swadeley commented:
My new test was missing vm from the parameter list. Now fixed.

 I am using the CLI wait_for_task function, not browser based search, to check that the upload task is complete and I can now check for package name updates in web UI. I discovered there is slight discrepancy in the time between CLI and Browser apart from the time zone difference (3 or 4 mins). This made my tests fail.  I had to include a 5 min margin for safety, and stripped off the seconds for good measure.

I updated the text before the line that takes the time stamp to make it clear what and why the line is like it is:

    +        # make a note of time for later CLI wait_for_tasks, and include
    +        # 5 mins margin of safety.
    +        timestamp = (datetime.utcnow() - timedelta(minutes=5)).strftime("%Y-%m-%d %H:%M")

swadeley commented:
robottelo]$ pytest --show-capture=no tests/foreman/ui/test_contenthost.py -k test_module_status_update_without_force_upload_package_profile
2019-08-19 18:56:17 - conftest - DEBUG - Registering custom pytest_configure

===================== test session starts ======================================
platform linux -- Python 3.6.6, pytest-4.6.3, py-1.8.0, pluggy-0.12.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /Downloads/robottelo
plugins: services-1.3.1, forked-1.0.2, xdist-1.29.0, cov-2.7.1, mock-1.10.4
collecting ... 2019-08-19 18:56:18 - conftest - DEBUG - BZ deselect is disabled in settings

collected 23 items / 22 deselected / 1 selected                                                                                                              

tests/foreman/ui/test_contenthost.py .                                                                        [100%]

========== 1 passed, 22 deselected, 4 warnings in 916.65 seconds ===================


robottelo]$ pytest --show-capture=no tests/foreman/ui/test_contenthost.py -k test_positive_end_to_end_bulk_update
2019-08-19 18:56:26 - conftest - DEBUG - Registering custom pytest_configure

====================== test session starts ======================================
platform linux -- Python 3.6.6, pytest-4.6.3, py-1.8.0, pluggy-0.12.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /Downloads/robottelo
plugins: services-1.3.1, forked-1.0.2, xdist-1.29.0, cov-2.7.1, mock-1.10.4
collecting ... 2019-08-19 18:56:26 - conftest - DEBUG - BZ deselect is disabled in settings

collected 23 items / 22 deselected / 1 selected                                                                                                              

tests/foreman/ui/test_contenthost.py .                                                                    [100%]

============== 1 passed, 22 deselected, 2 warnings in 785.25 seconds ================

swadeley commented:
Hello

I ran a test against the whole file. My tests pass again, but one other fails:

 robottelo]$ pytest --show-capture=no tests/foreman/ui/test_contenthost.py
2019-08-19 20:10:55 - conftest - DEBUG - Registering custom pytest_configure

========================== test session starts ==================================
platform linux -- Python 3.6.6, pytest-4.6.3, py-1.8.0, pluggy-0.12.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /Downloads/robottelo
plugins: services-1.3.1, forked-1.0.2, xdist-1.29.0, cov-2.7.1, mock-1.10.4
collecting ... 2019-08-19 20:10:55 - conftest - DEBUG - BZ deselect is disabled in settings

collected 23 items                                                                                                                                           

tests/foreman/ui/test_contenthost.py F......................                                                     [100%]

============================ FAILURES ===================================
_________________________ test_positive_end_to_end _______________________________

    >           assert actual_repos == expected_repos
    E           AssertionError: assert {'Red Hat Ans...HNySIOcOYJwX'} == {'Red Hat  Ansi...HNySIOcOYJwX'}
    E             Extra items in the left set:
    E             'sjDFjHYZkmDNUAs'
    E             Use -v to get the full diff

tests/foreman/ui/test_contenthost.py:197: AssertionError

========== 1 failed, 22 passed, 45 warnings in 7830.37 seconds =====

omkarkhatavkar commented:
@swadeley It looks good to me.

vijay8451 commented on tests/foreman/ui/test_contenthost.py:
```diff
@@ -256,22 +255,24 @@ def test_positive_end_to_end_bulk_update(session, vm):
                          ' and started_at >= "{}"'.format(
                              host[0].id, timestamp),
-            search_rate=10, max_tries=10,
+            search_rate=30, max_tries=10,
```
I think `15-20` are good for search_rate rate and would help to save some time , what do you think ?

swadeley replied:
OK, I will change to every 15 seconds.

 I checked with evgeni, the yum package_upload plug-in from the katello-hosts-tools.rpm does the upload and it just does it "post transaction", no specific time delay or polling interval. I thought there was a few mins delay.

swadeley replied:
OK, done that. Bare in mind for a test that is going to pass, this only makes 15 seconds difference. For a test that is going to fail, its 2.5 mins difference.

vijay8451 replied:
yeah it would save some seconds , as we have `max_tries` as well so we may utilize that if requires to increase , What do you think ?

swadeley replied:
its OK, I made the change after consulting evgeni as to the expected profile upload time delay. 

Please merge and squash the commits. If it fails sometimes due to slow VMs or test system we can extend it.

vijay8451 commented on tests/foreman/ui/test_contenthost.py:
```diff
@@ -915,6 +916,80 @@ def test_module_status_update_from_content_host_to_satellite(session, vm_module_
+                         ' and resource_id = {}'
+                         ' and started_at >= "{}"'.format(
+                             host[0].id, timestamp),
+            search_rate=30, max_tries=10,
```
same as above.

swadeley replied:
done, see above

swadeley commented:
Thank you all.
